ragnarok_packets/src/position.rs

```diff
@@ -5,11 +5,12 @@ use ragnarok_bytes::{ByteReader, ConversionResult, FromBytes, ToBytes};
 pub struct WorldPosition {
     pub x: usize,
     pub y: usize,
+    pub direction: usize,
```

Comment:
Couldn't we create a type for this? I'm not sure which number has which direction, so you would need to double-check that.

```rust
#[derive(Debug, Copy, Clone, ByteConvertable)]
#[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
pub enum Direction {
    N = 0,
    NE = 1,
    E = 2,
    SE = 3,
    S = 4,
    SW = 5,
    W = 6,
    NW = 7,
}

impl From<Direction> for usize {
    fn from(value: Direction) -> Self {
        value as usize
    }
}

impl From<usize> for Direction {
    fn from(value: usize) -> Self {
        match value {
            1 => Direction::NE,
            2 => Direction::E,
            3 => Direction::SE,
            4 => Direction::S,
            5 => Direction::SW,
            6 => Direction::W,
            7 => Direction::NW,
            0 | _ => Direction::N,
        }
    }
}
```

Reply:
The `From<usize>` will be optimized to:

```asm
mov     rax, rdi
and     al, 7
ret
```

Reply:
Not sure if I used into() in the way it should be used.